js/atoms/src/utils/parse_placeholder.ts

```diff
@@ -0,0 +1,21 @@
+const RE_HEADING = /^#\s*(.+)$/m;
+
+export function inject(text: string): [string | false, string | false] {
+	const trimmed_text = text.trim();
+
+	const heading_match = trimmed_text.match(RE_HEADING);
+	const _heading = heading_match ? heading_match[1].trim() : false;
+
+	const heading_end_index = heading_match
+		? trimmed_text.indexOf("\n", heading_match.index)
```

Comment:
It is safer to use the actual length of the matched text (plus the two characters for the hash and space) because we cannot guarantee that there will be a new line character.